File: leetcode-solutions/1642-water-bottles/solution.py
class Solution(object):
    def numWaterBottles(self, numBottles, numExchange):
        """
        :type numBottles: int
        :type numExchange: int
        :rtype: int
        """
        drinked, empty, full = 0, 0, numBottles
        while full > 0:
            drinked += full
            empty += full
            full = empty // numExchange
            empty = empty % numExchange
        return drinked

File: leetcode-solutions/1642-water-bottles/solution.java
class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int drinked = 0, empty = 0, full = numBottles;
        while(full > 0) {
            drinked += full;
            empty += full;
            full = empty / numExchange;
            empty %= numExchange;
        }
        return drinked;
    }
}
